jetty-http/src/test/java/org/eclipse/jetty/http/GZIPContentDecoderTest.java

```diff
@@ -351,4 +358,97 @@ public void testBigBlockWithExtraBytes() throws Exception
         assertTrue(buffer.hasRemaining());
         assertEquals(data2, StandardCharsets.UTF_8.decode(buffer).toString());
     }
+
+    // Signed Integer Max
+    final long INT_MAX = Integer.MAX_VALUE;
+
+    // Unsigned Integer Max == 2^32
+    final long UINT_MAX = 0xffffffffL;
+
+    @ParameterizedTest
+    @ValueSource(longs = {INT_MAX, INT_MAX + 1, UINT_MAX, UINT_MAX + 1})
+    public void testLargeGzipStream(long origSize) throws IOException
+    {
+        final int BUFSIZE = (1 * 1024 * 1024); // 1MB
+
+        // Create a buffer to use over and over again to produce the uncompressed input
+        byte[] cbuf = "0123456789ABCDEFGHIJKLMOPQRSTUVWXYZ".getBytes(StandardCharsets.UTF_8);
+        byte[] buf = new byte[BUFSIZE];
```

Comment:
I don't think you need `buf` at all. 
Write `cbuf` many times, up to `origSize`, and you have the same data and much less code.

Reply:
and a *REALLY SLOW* test.
I did that originally, and the test for Integer.MAX_VALUE took nearly 2 minutes.
But in this way it takes only 18 seconds.

Reply:
Wow, so what if you make it 16 MiB? Is it any faster?

Reply:
I tried that.
I can't get it to run faster with a greater than 1MB buffer.
I suspect I switched from being slowed down by the buffer size to being slowed down just due to the gzip encoding itself.